src/lib/clean.c

```diff
@@ -57,61 +54,42 @@ static bool aa_hook_context_has_cache(AaHookContext *self, const char *d_name)
  */
 bool aa_hook_context_clean_cache(AaHookContext *self)
 {
-        autofree(DIR) *dir = NULL;
-        struct dirent *ent = NULL;
-        bool ret = false;
-        int fd = 0;
+        FTSENT* curr_entry = NULL;
+        char* path[] = { self->cache_dir, NULL };
+        FTS* file_system = fts_open(path, FTS_PHYSICAL | FTS_NOSTAT, NULL);
 
-        if (aa_unlikely(!self)) {
+        if (file_system == NULL) {
                 return false;
         }
 
-        /* If the directory doesn't exist yet, it isn't fatal */
-        dir = opendir(self->cache_dir);
-        if (!dir) {
-                if (errno == ENOENT) {
-                        return true;
+        bool ret = true;
```

Comment:
Should be declared at the top. I know it seems archaic, but let's keep the style the same.